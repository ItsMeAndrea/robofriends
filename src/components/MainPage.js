import React, { useEffect } from "react";
import CardList from "../components/CardList";
import SeachBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import ErrorBoundry from "../components/ErrorBoundry";
import Header from "../components/Header";
import "./MainPage.css";

export const filteredRobots = (robots, search) => {
  return robots.filter((robot) => {
    return robot.name.toLowerCase().includes(search.toLowerCase());
  });
};

const MainPage = ({
  search,
  onSearchChange,
  onRequestRobots,
  robots,
  isPending,
}) => {
  useEffect(() => {
    onRequestRobots();
  }, []);

  return isPending ? (
    <h1 className="tc">Loading...</h1>
  ) : (
    <div className="tc">
      <Header />
      <SeachBox searchChange={onSearchChange} />
      <Scroll>
        <ErrorBoundry>
          <CardList robots={filteredRobots(robots, search)} />
        </ErrorBoundry>
      </Scroll>
    </div>
  );
};

export default MainPage;
