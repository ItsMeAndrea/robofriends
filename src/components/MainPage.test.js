import React from "react";
import { mount } from "enzyme";
import MainPage, { filteredRobots } from "./MainPage";

let wrapper;
const robotsarray = [
  { name: "joe", email: "joe@gmail.com" },
  { name: "mark", email: "joe@gmail.com" },
  { name: "john", email: "joe@gmail.com" },
];

beforeEach(() => {
  const mockProps = {
    robots: [],
    search: "",
    onRequestRobots: jest.fn(),
    isPending: false,
  };
  wrapper = mount(<MainPage {...mockProps} />);
});

it("renders MainPage", () => {
  expect(wrapper).toMatchSnapshot();
});

it("filters robots correctly", () => {
  expect(filteredRobots(robotsarray, "joe")).toEqual([
    {
      name: "joe",
      email: "joe@gmail.com",
    },
  ]);
});
