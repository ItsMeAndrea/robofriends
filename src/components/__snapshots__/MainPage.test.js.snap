// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage 1`] = `
<MainPage
  isPending={false}
  onRequestRobots={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  robots={Array []}
  search=""
>
  <div
    className="tc"
  >
    <Header>
      <h1
        className="f1"
      >
        RoboFriends
      </h1>
    </Header>
    <SearchBox>
      <div
        className="pa2"
      >
        <input
          aria-label="searchbox"
          className="pa3 ba b--green bg-lightest-blue"
          placeholder="search robots"
          type="search"
        />
      </div>
    </SearchBox>
    <Scroll>
      <div
        style={
          Object {
            "height": "500px",
            "overflowY": "scroll",
          }
        }
      >
        <ErrorBoundry>
          <CardList
            robots={Array []}
          >
            <div />
          </CardList>
        </ErrorBoundry>
      </div>
    </Scroll>
  </div>
</MainPage>
`;
